server_admin/topics/authentication/x509.adoc

```diff
@@ -209,6 +209,61 @@ image:images/x509-directgrant-flow.png[]
 
 image:images/x509-directgrant-flow-bindings.png[]
 
+==== Client certificate lookup
+
+When user sends HTTP requests directly to {project_name} server, it's the {appserver_name} undertow subsystem, which will establish the SSL handshake and extracts client certificate. Client certificate will be then saved to the attribute `javax.servlet.request.X509Certificate` of the HTTP request as specified in the servlet specification. {project_name} X509 authenticator will be then able to lookup the certificate from this attribute.
+
+However when {project_name} listens to HTTP requests behind the loadbalancer or reverse proxy, it might be the proxy server, which extracts the client certificate and establish the mutual SSL connection. Reverse proxy usually put the authenticated client certificate to the HTTP header of the underlying request and forwards it to the backend {project_name} server. In this case, {project_name} must be able to lookup the X.509 certificate chain from the HTTP headers instead of from the attribute of HTTP request like done for Undertow.
+
+If {project_name} is behind the reverse proxy, you usually need to configure alternative provider of the `x509cert-lookup` SPI in `{project_dirref}/standalone/configuration/standalone.xml` file. Besides the `default` provider, which lookups the certificate from the HTTP header, we have 2 builtin providers: `haproxy` and `apache`.
```

Comment:
If {project_name} is behind a reverse proxy, you usually need to configure alternative provider of the `x509cert-lookup` SPI in `{project_dirref}/standalone/configuration/standalone.xml`. Along with the `default` provider, which looks up the certificate from the HTTP header, we also have two additional built-in providers: `haproxy` and `apache`, which are described next.